src/libsync/propagateupload.h

```diff
@@ -155,10 +160,87 @@ class PollJob : public AbstractNetworkJob {
 };
 
 /**
- * @brief The PropagateUploadFile class
+ * @brief The PropagateUploadFileCommon class is the code common between all chunking algorithms
  * @ingroup libsync
+ *
+ * State Machine:
+ *
+ *   +---> start()  --> (delete job) -------+
+ *   |                                      |
+ *   +--> slotComputeContentChecksum()  <---+
+ *                   |
+ *                   v
+ *    slotComputeTransmissionChecksum()
+ *         |
+ *         v
+ *    slotStartUpload()  -> doStartUpload()
+ *                                  .
+ *                                  .
+ *                                  v
+ *        finalize() or abortWithError()  or startPollJob()
+ */
+class PropagateUploadFileCommon : public PropagateItemJob {
+    Q_OBJECT
+
+protected:
+    QElapsedTimer _duration;
+    QVector<AbstractNetworkJob*> _jobs; /// network jobs that are currently in transit
+    bool _finished; /// Tells that all the jobs have been finished
+    bool _deleteExisting;
+
+    // measure the performance of checksum calc and upload
+    Utility::StopWatch _stopWatch;
```

Comment:
Maybe we can drop the stopwatch here, it was more to proof that checksum calculation does not take long, but these discussions are gone thankfully.